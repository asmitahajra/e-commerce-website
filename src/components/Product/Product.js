import PropTypes from 'prop-types';
import React from 'react';
// import './Product.css';
import Counter from '../Counter/Counter';
import styles from './Product.module.css';
// import { ThemeContext } from '../../ThemeContext';

const Product = ({ product, onIncrement, onDecrement }) => {
  // const Theme = useContext(ThemeContext);
  // const currentTextColor = Theme.color;
  // const { product, onIncrement, onDecrement } = props;
  const {
    // eslint-disable-next-line react/prop-types
    name, price, count, image,
  } = product;
  return (
    <div data-testid="counter" className={styles.product_card_light}>
      <div className="details">
        <img data-testid="card-img" src={image} alt={name} className={styles.image} />
        {/* <p>{company}</p> */}
        <p><b>{name}</b></p>
        <p>1 kg</p>
      </div>
      <div className={styles.check}>
        <p>
          <b>
            MRP
            {' '}
            {price}
            /-
          </b>
        </p>
        <Counter count={count} onIncrement={onIncrement} onDecrement={onDecrement} />
      </div>
    </div>
  );
};

Product.propTypes = {
  product: PropTypes.shape({
    id: PropTypes.number.isRequired,
    // company: PropTypes.string.isRequired,
    // image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    count: PropTypes.number.isRequired,
  }).isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};

export default Product;
