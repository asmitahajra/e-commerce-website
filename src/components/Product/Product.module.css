.product_card_light {
    box-shadow: 10px 10px 5px grey;
    padding: 0.5rem;
    border: black solid 1px;
    border-radius: 0.9rem;
    margin: 0.9rem;
    max-width: fit-content;
}

.product_card_dark {
    box-shadow: 10px 10px 5px grey;
    padding: 0.5rem;
    border: black solid 1px;
    border-radius: 0.9rem;
    margin: 0.9rem;
    max-width: fit-content;
    background-color: black;
}

.image {
    width: 10rem;
    height: 10rem;
}

.check {
    padding: 0.5rem;
    background-color: lightgray;
}