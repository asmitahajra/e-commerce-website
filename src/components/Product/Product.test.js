/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render, screen } from '@testing-library/react';
import Product from './Product';
// import { Theme, ThemeContext } from '../../ThemeContext';

describe(Product.name, () => {
  const mockProduct = {
    product: {
      id: 1,
      name: 'apple',
      price: 20,
      count: 1,
      image: 'apple.jpg',
      company: 'FRSHO',
    },
    onIncrement: jest.fn(),
    onDecrement: jest.fn(),
  };
  test('sanity', () => {
    expect(1 + 1).toBe(2);
  });

  test('should match snapshot', () => {
    const { container } = render(<Product {...mockProduct} />);
    expect(container).toMatchSnapshot();
  });

  test('should display company name', () => {
    render(<Product {...mockProduct} />);
    screen.getByText('FRSHO');
  });

  test('should display product name', () => {
    render(<Product {...mockProduct} />);
    screen.getByText('apple');
  });

  test('should display product quantity', () => {
    render(<Product {...mockProduct} />);
    screen.getByText('1 kg');
  });

  test('should display product price', () => {
    render(<Product {...mockProduct} />);
    screen.getByText('MRP 20/-');
  });

  test('should display product image', () => {
    render(<Product {...mockProduct} />);
    screen.getByTestId('card-img');
  });

  test('should render Counter', () => {
    render(<Product {...mockProduct} />);
    screen.getByTestId('counter');
  });
});
