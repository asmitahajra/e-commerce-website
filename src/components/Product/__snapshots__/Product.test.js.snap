// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should match snapshot 1`] = `
<div>
  <div
    class="product_card_light"
    data-testid="counter"
  >
    <div
      class="details"
    >
      <img
        alt="apple"
        class="image"
        data-testid="card-img"
        src="apple.jpg"
      />
      <p>
        FRSHO
      </p>
      <p>
        <b>
          apple
        </b>
      </p>
      <p>
        1 kg
      </p>
    </div>
    <div
      class="check"
    >
      <p>
        <b>
          MRP
           
          20
          /-
        </b>
      </p>
      <div
        class="counter"
      >
        <button
          style="background: white; color: black;"
          type="button"
        >
          +
        </button>
        <p
          class="inBasket"
          style="background: gray;"
        >
          1
           
          in Basket
        </p>
        <button
          style="background: white; color: black;"
          type="button"
        >
          -
        </button>
      </div>
    </div>
  </div>
</div>
`;
